packages/preset-typescript/index.js

```diff
@@ -1,6 +1,8 @@
-function webpack(webpackConfig = {}, options = {}) {
+async function webpack(webpackConfig = {}, options = {}) {
   const { module = {}, resolve = {} } = webpackConfig;
-  const { tsLoaderOptions, tsDocgenLoaderOptions, include } = options;
+  const { tsLoaderOptions, tsDocgenLoaderOptions, include, presets } = options;
+
+  const loaderOptions = await presets.apply('tsLoaderOptions', tsLoaderOptions, { presets });
```

Comment:
I don't fully understand this line. Where is presets defined? Is there an example usage somewhere I can look at?

Reply:
presets is destructured from options, this is how preset work.